bin/run.py

```diff
@@ -169,6 +169,47 @@ def validate_format(
 
             break
 
+        if not config.args.skip_testcase_sanity_checks and success and not bad_testcase :
+            if validator_type == 'input_format' and self.in_path.exists():
+                with self.in_path.open() as in_file:
+                    def invalid(byte):
+                        if byte == ord('\n'): return False;
+                        if byte >= 0x20 and byte < 0x7F : return False;
+                        return True
+
+                    bytes = in_file.buffer.read()
+                    if any(invalid(b) for b in bytes):
+                        bar.warn('Testcase contains unexpected characters but was accepted!')
+                    else:
+                        input = bytes.decode(encoding="ascii")
+                        if re.search(r"^\s", input):
+                            bar.warn('Testcase starts with whitespace but was accepted!')
+                        elif re.search(r"\s\s", input):
+                            bar.warn('Testcase contains consecutive whitespace characters but was accepted!')
+                        elif not re.search(r"\S", input):
+                            bar.warn('Testcase is empty but was accepted!')
+                        elif not re.search(r"\n$", input):
+                            bar.warn('Testcase does not end with a newline but was accepted!')
+                        elif len(bytes) > 20_000_000_000:
+                            bar.warn('Testcase is larger than 20Mb!')
+
+            if validator_type == 'output_format' and self.ans_path.exists():
+                with self.ans_path.open() as in_file:
+                    def invalid(byte):
+                        if byte == ord('\t'): return False;
```

Comment:
Can you add a comment why this is different that the input invalid bytes?
I.e.: input allows only spaces and newlines, while output is a bit more lenient.

Also, `def ...` inside a loop is relatively slow in python. (Not that this is critical code, but anyway.) Better to define a separate function `invalida(byte, input/output)`